util/filesort/filesort_test.go

```diff
@@ -177,51 +167,51 @@ func (s *testFileSortSuite) TestMultipleFiles(c *C) {
 	)
 
 	tmpDir, err = os.MkdirTemp("", "util_filesort_test")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	fsBuilder := new(Builder)
 
 	// Test for basic function.
 	_, err = fsBuilder.Build()
-	c.Assert(err.Error(), Equals, "StatementContext is nil")
+	require.Equal(t, "StatementContext is nil", err.Error())
 	fsBuilder.SetSC(sc)
 	_, err = fsBuilder.Build()
-	c.Assert(err.Error(), Equals, "key size is not positive")
+	require.Equal(t, "key size is not positive", err.Error())
 	fsBuilder.SetDesc(byDesc)
 	_, err = fsBuilder.Build()
-	c.Assert(err.Error(), Equals, "mismatch in key size and byDesc slice")
+	require.Equal(t, "mismatch in key size and byDesc slice", err.Error())
 	fsBuilder.SetSchema(keySize, valSize)
 	_, err = fsBuilder.Build()
-	c.Assert(err.Error(), Equals, "buffer size is not positive")
+	require.Equal(t, "buffer size is not positive", err.Error())
```

Comment:
You can use `EqualError` here also.

Reply:
ditto others

Reply:
`EqualError` expects the second argument to be  of `error` type and third argument as `string`.

`err.Error()` is of `string` type. So since both second argument i.e "buffer size is not positive" and third argument, err.Error() are of `string` type, I'm using `Equal`.

Reply:
If you take a look at how `require.EqualError` implements, you will found

```go
require.Equal(t, "buffer size is not positive", err.Error())
// equals to
require.EqualError(t, err, "StatementContext is nil")
```

... with extra check `err` is not `nil` so you won't panic and IDE like GoLand won't complain about nil pointer potential.